### PR TITLE
fix: percent to ratio

### PR DESCRIPTION
**Pull request by e-sailing**
tank level has the unit ratio ( 0 to 1) but was used as percentage ( 0
to100 )

### CODE REVIEW

**Comment by tkurki** [comment]
Most of the test data used in the tests is based on real world samples and should be replaced only with data from the real world and not fixed manually.

Analyzer outputs humidity as percentage 0-100, not ratio 0-1:

```
$ echo '2015-01-15-16:13:59.866,5,130311,130,255,8,89,41,a9,6d,1e,4a,fa,03' | analyzer -si -json | json
INFO 2017-08-03T07:14:08.186Z [analyzer] New PGN 130311 for device 130 (heap 9487 bytes)
{
  "timestamp": "2015-01-15-16:13:59.866",
  "prio": 5,
  "src": 130,
  "dst": 255,
  "pgn": 130311,
  "description": "Environmental Parameters",
  "fields": {
    "SID": 137,
    "Temperature Source": "Outside Temperature",
    "Humidity Source": "Outside",
    "Temperature": 280.73,
    "Humidity": 75.896,
    "Atmospheric Pressure": 101800
  }
}
```

Why did you tweak humidity and the other input values?

**Comment by e-sailing** [comment]
humidity:
Sorry. My simulation was wrong (I changed to much things at the same time).
Now I restored the json file and made changes 0-100 -> 0-1 in 130311.js.
(It would be better to make a test file for 130311_environment.js.)

tweak other input values:
Some doubles / 100 have a result ending with ...0000001
I don't know how to call a function like "should.be.closeTo" in this situation.
( delta.updates[0].values[0].value.longitude.should.be.closeTo(23.432133, 0.000005); )
So I used (tweaked) values that can devided by 100 with exactly the same result.
I can make a cleaner solution when you point me into the right direction.

**Comment by tkurki** [comment]
The test fails now because your 130311 Humidity change causes the code to emit NaN's: you are coercing undefined's to Number. You need to return undefined if there is no data or add 
```
    filter: function (n2k) {
      return n2k.fields['Humidity']
    },
```

**Comment by tkurki** [comment]
Instead of manually tweaking the numbers to match I think `.should.be.closeTo` per your comment would be a better way.

```
    tree.tanks.freshWater['0'].currentLevel.value.should.be.closeTo(
      0.88596,
      0.000005
    )
```

**Comment by tkurki** [comment]
Please format your changed code with `npm run format`.

File: pgns/130311.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ module.exports = [
         '.humidity'
       )
     },
+	filter: function (n2k) {
+      return n2k.fields['Humidity']
```
Comment (tkurki) [inline pgns/130311.js]: This will be falsy if the value is 0. Not a huge deal, as it probably never is, but a typeof != 'undefined' test would be better.

**Comment by e-sailing** [comment]
I hope
return typeof n2k.fields['Humidity'] !== 'undefined'
is the right answer.
I'm not sure, if we also need a filter on 127505.

**Pull request merged**